packages/camera/camera_android_camerax/lib/src/android_camera_camerax.dart

```diff
@@ -13,9 +18,67 @@ class AndroidCameraCameraX extends CameraPlatform {
     CameraPlatform.instance = AndroidCameraCameraX();
   }
 
+  ProcessCameraProvider? processCameraProvider;
+  CameraSelector? backCameraSelector;
+  CameraSelector? frontCameraSelector;
+
+
   /// Returns list of all available cameras and their descriptions.
   @override
   Future<List<CameraDescription>> availableCameras() async {
-    throw UnimplementedError('availableCameras() is not implemented.');
+    final List<CameraDescription> cameraDescriptions = <CameraDescription>[];
+
+    processCameraProvider ??= await ProcessCameraProvider.getInstance();
```

Comment:
If you keep the setter for this, I'd use it here for consistency. You can move the null check to that method, too.